GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -873,9 +873,15 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
             // The information from step 1. is therefore protected with a semaphore.
             // This local semaphore instance is bound in closures below.
             System.Threading.SemaphoreSlim semaphoreCurrentCommit = new(initialCount: 0, maxCount: 1);
-            _isRefreshingRevisions = true;
 
             // Protected by semaphoreCurrentCommit
+
+            _isRefreshingRevisions = true;
+            //// TODO: What if we do not reach the end of OnRevisionReadCompleted(), where _isRefreshingRevisions is reset?!?
+
+            System.Threading.CancellationToken cancellationToken = _refreshRevisionsSequence.Next();
```

Comment:
[nit] please use `using System.Threading;`
```suggestion
            CancellationToken cancellationToken = _refreshRevisionsSequence.Next();
```

Reply:
OK, (I just moved the line)

Reply:
fixed